velox/functions/lib/Re2Functions.cpp

```diff
@@ -666,6 +783,112 @@ FOLLY_ALWAYS_INLINE static bool isAsciiArg(
+    return false;
+  }
+
+  // Search the specified literal pattern in the input.
+  auto inputView = std::string_view(input);
+  auto patternStrHeader = patternMetadata.fixedPattern().c_str();
+  // The index in the input we are matching against.
+  auto currentMatchingOffsetInInput = 0;
+  auto searchLiteralPattern =
+      [&](const SubPatternMetadata& subPattern) -> int32_t {
+    auto index = inputView.find(
+        patternStrHeader + subPattern.start,
+        currentMatchingOffsetInInput,
+        subPattern.length);
+    if (index == std::string::npos) {
+      return -1;
+    }
+
+    return index;
+  };
+
+  // The index of the sub literal pattern we are matching.
+  auto currentPatternIndex =
+      (patternMetadata.subPatterns()[0].kind == kLiteralString ? 0 : 1);
+  auto currentSubPattern = patternMetadata.subPatterns()[currentPatternIndex];
+  auto end = input.size();
+  std::unique_ptr<RelaxedPatternMatchResult> matchResult;
+  while (currentMatchingOffsetInInput < end) {
+    // Search current literal pattern to find out where to start matching
+    // the whole pattern.
+    currentMatchingOffsetInInput = searchLiteralPattern(currentSubPattern);
+    // If any literal pattern is not found, fails the matching.
+    if (currentMatchingOffsetInInput < 0) {
+      return false;
+    }
+
+    // Match the right part of the pattern(if there is a right part).
+    if (!isRightMostSubPattern(patternMetadata, currentPatternIndex)) {
+      matchResult = std::make_unique<RelaxedPatternMatchResult>(
+          matchRelaxedFixedForward<isAscii>(
+              input,
+              patternMetadata,
+              currentMatchingOffsetInInput + currentSubPattern.length,
+              currentPatternIndex + 1));
+    }
+
+    // Match the left part of the pattern(if there is a left part).
+    // We only match the left part if:
+    // - Current pattern is the right most.
+    // - We successfully matched the right part of the pattern.
+    if (isRightMostSubPattern(patternMetadata, currentPatternIndex) ||
+        (matchResult->match() && currentPatternIndex > 0)) {
+      matchResult = std::make_unique<RelaxedPatternMatchResult>(
+          matchRelaxedFixedBackward<isAscii>(
+              input,
+              patternMetadata,
+              currentMatchingOffsetInInput - 1,
+              currentPatternIndex - 1));
+    }
+
+    // Both the left and right part of the pattern match, the whole pattern
+    // matches.
+    if (matchResult->match()) {
+      return true;
+    }
+
+    if (matchResult->isInputTooShort()) {
+      return false;
+    }
+
+    // Search the mismatch literal pattern to find the location where we can
+    // start matching.
+    currentPatternIndex = matchResult->failingPatternIndex();
+    currentSubPattern = patternMetadata.subPatterns()[currentPatternIndex];
+    // The mismatch pattern is not a literal means the input is not long
+    // enough.
+    if (currentSubPattern.kind != kLiteralString) {
```

Comment:
@xumingming Any particular reason this case is not covered by matchResult->isInputTooShort() check above?